utils/logalloc.cc

```diff
@@ -704,27 +704,27 @@ class segment_store {
     segment_store() : _segments(max_segments()) {
         _segment_indexes.reserve(max_segments());
     }
-    segment* segment_from_idx(size_t idx) const {
+    segment* segment_from_idx(size_t idx) const noexcept {
         assert(idx < _segments.size());
         return _segments[idx];
     }
-    size_t idx_from_segment(segment* seg) {
+    size_t idx_from_segment(segment* seg) noexcept {
         // segment 0 is a marker for no segment
         auto i = _segment_indexes.find(seg);
         if (i == _segment_indexes.end()) {
             return 0;
         }
         return i->second;
     }
-    size_t new_idx_for_segment(segment* seg) {
+    size_t new_idx_for_segment(segment* seg) noexcept {
         auto i = find_empty();
```

Comment:
Can we mark find_empty as noexcept as well since we're not using ExecutionPolicy and pointer comparison won't throw?